### PR TITLE
book: Add dev guide page about Rust dependencies

### PR DESCRIPTION
**Pull request by str4d**

### CODE REVIEW

File: doc/book/src/dev/rust.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+# patch goes into a [patch.crates-io] section.
+orchard = "0.0"
+...
+
+[patch.crates-io]
+# Comment out any existing patch, if present.
+# orchard = { git = "https://github.com/zcash/orchard.git", rev = "..." }
+
+# Add this patch:
+orchard = { path = "../relative/path/to/orchard" }
```
Comment (LarryRuane) [inline doc/book/src/dev/rust.md:51]: It seems to work with an absolute path, but if it's safer or better to use a relative path, we can leave this as is.

File: doc/book/src/dev/rust.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+After doing so, you can add a new dependency as follows:
+
+1. Add the new dependency to `Cargo.toml`.
+2. Run `cargo check` to update the `Cargo.lock` file.
+3. Commit `Cargo.toml` and `Cargo.lock`.
+
+## Using a local Rust dependency
+
+During development, you can use a locally checked out version of a dependency
+by applying a [`cargo` patch](https://doc.rust-lang.org/cargo/reference/overriding-dependencies.html#the-patch-section).
```
Comment (LarryRuane) [inline doc/book/src/dev/rust.md:34]: Verified this link 👍

**Review by LarryRuane — APPROVED** [review]
ACK 292f83050997cb7e4d803871593ef636b49d7720

**Review by daira — APPROVED** [review]
utACK

File: doc/book/src/dev/rust.md (hunk 1 above)
Comment (str4d) [reply doc/book/src/dev/rust.md:51]: It will work with either.

File: doc/book/src/dev/rust.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+# This dependency is listed with a version, meaning it comes from crates.io; the
+# patch goes into a [patch.crates-io] section.
+orchard = "0.0"
+...
+
+[patch.crates-io]
+# Comment out any existing patch, if present.
+# orchard = { git = "https://github.com/zcash/orchard.git", rev = "..." }
+
+# Add this patch:
```
Comment (str4d) [inline doc/book/src/dev/rust.md:50]:
```suggestion
# Add this patch (both relative and absolute paths work):
```

**Comment by str4d** [comment]
@zkbot r+

**Comment by str4d** [comment]
Transient CI failure (being fixed in #5222).

@zkbot retry

**Pull request merged**